### DISCUSSION TITLE
Third party providers session null

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3492] [General]
Hi everyone!
When i use a Third party provider for authentication it's gives me `session = null`

And that's because of the redirection 

So what about the use of something like `Firebase signInWithPopup() window`?

If there is any available solution please tell me 😊.

Thanks 💛

**Comment by zakaria-chahboun** [comment]
At this moment i did this solution https://github.com/supabase/supabase/discussions/2842#discussioncomment-1486453
in SvelteKit (but the the concept is one)

**Comment by fergusmeiklejohn** [comment]
I think the cleanest solution for setting a cookie is to do all the auth on the server. Bouncing off the client is inelegant and prone to bugs where the data isn't available yet when the setCookie function is called

**Reply by zakaria-chahboun** [reply]
the problem with supabase at this moment that we can't do the token auto refresh in the server side!
